src/main/java/org/mule/tools/apikit/input/MuleConfigParser.java

```diff
@@ -6,99 +6,61 @@
  */
 package org.mule.tools.apikit.input;
 
-import org.apache.maven.plugin.logging.Log;
 import org.jdom2.Document;
-import org.jdom2.input.SAXBuilder;
-import org.jdom2.input.sax.XMLReaders;
-import org.mule.tools.apikit.input.parsers.APIKitConfigParser;
 import org.mule.tools.apikit.input.parsers.APIKitFlowsParser;
 import org.mule.tools.apikit.input.parsers.APIKitRoutersParser;
-import org.mule.tools.apikit.input.parsers.HttpListener4xConfigParser;
-import org.mule.tools.apikit.model.API;
-import org.mule.tools.apikit.model.APIFactory;
-import org.mule.tools.apikit.model.APIKitConfig;
-import org.mule.tools.apikit.model.HttpListener4xConfig;
-import org.mule.tools.apikit.model.ResourceActionMimeTypeTriplet;
-
-import java.io.File;
-import java.io.InputStream;
-import java.util.Collection;
+import org.mule.tools.apikit.model.*;
+
 import java.util.HashMap;
 import java.util.HashSet;
+import java.util.List;
 import java.util.Map;
 import java.util.Set;
 
-import static java.util.Map.Entry;
 
 public class MuleConfigParser {
 
   private Set<ResourceActionMimeTypeTriplet> entries = new HashSet<>();
-  private Map<String, API> includedApis = new HashMap<>();
+  private Map<String, ApikitMainFlowContainer> includedApis = new HashMap<>();
   private Map<String, APIKitConfig> apikitConfigs = new HashMap<>();
   private final APIFactory apiFactory;
-  private final Log log;
 
-  public MuleConfigParser(Log log, APIFactory apiFactory) {
+  public MuleConfigParser(APIFactory apiFactory) {
     this.apiFactory = apiFactory;
-    this.log = log;
   }
 
-  public MuleConfigParser parse(Set<String> apiFilePaths, Map<File, InputStream> streams) {
-    Map<File, Document> configurations = createDocuments(streams);
-
-    for (Entry<File, Document> fileStreamEntry : configurations.entrySet()) {
-      Document document = fileStreamEntry.getValue();
-      File file = fileStreamEntry.getKey();
-      parseConfigs(file, document, apiFilePaths);
+  public MuleConfigParser parse(String apiLocation, List<MuleConfig> muleConfigs) {
+    for (MuleConfig config : muleConfigs) {
+      parseConfigs(config);
     }
 
-    for (Entry<File, Document> fileStreamEntry : configurations.entrySet()) {
-      Document document = fileStreamEntry.getValue();
-      File file = fileStreamEntry.getKey();
-      parseApis(file, document, apiFilePaths);
+    for (MuleConfig config : muleConfigs) {
```

Comment:
Unfortunately, the current logic of the APIKitRoutersParser doesn't allow it. First it has to go over all the mule configs and _parse_ its configs (apikit configs and http listeners configs) and then go over the mule configs again to parse all the apikit's main flows (I guess it was made in that way because the apikit configs and https listener configs can be declared in another mule config).